### PR TITLE
fix: BLS checkMalleable protection

### PR DESCRIPTION
**Pull request by ogabrielides**
## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — DISMISSED** [review]
makes sense I guess, utACK

File: src/bls/bls.h [hunk 1]
```diff
@@ -190,7 +190,18 @@ class CBLSWrapper
...
         if (checkMalleable && !CheckMalleable(vecBytes, specificLegacyScheme)) {
-            throw std::ios_base::failure("malleable BLS object");
+            // If CheckMalleable failed with specificLegacyScheme, we need to try again with the opposite scheme.
+            // Probably we received the BLS object sent with legacy scheme, but in the meanwhile the fork activated.
+            SetByteVector(vecBytes, !specificLegacyScheme);
+            if (!CheckMalleable(vecBytes, !specificLegacyScheme)) {
+                // Both attempts failed
+                throw std::ios_base::failure("malleable BLS object");
+            }
+            else {
```
Comment (PastaPastaPasta) [inline src/bls/bls.h:200]:
```suggestion
            } else {
```

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Pull request merged**